src/ring_theory/derivation.lean

```diff
@@ -440,3 +447,97 @@ begin
 end
 
 end to_square_zero
+
+section derivation_module
+
+open_locale tensor_product
+
+variables (R S : Type*) [comm_ring R] [comm_ring S] [algebra R S]
+
+/-- The kernel of the map `(𝟙, 𝟙) : S ⊗[R] S →ₐ[R] S`. -/
+abbreviation derivation_module.ideal : ideal (S ⊗[R] S) :=
+ring_hom.ker (tensor_product.lmul' R : S ⊗[R] S →ₐ[R] S)
+
+variable {S}
+
+lemma derivation_module.one_smul_sub_smul_one_mem_ideal (a : S) :
+  (1 : S) ⊗ₜ[R] a - a ⊗ₜ[R] (1 : S) ∈ derivation_module.ideal R S :=
+by simp [ring_hom.mem_ker]
+
+variables {R}
+
+variables {M : Type*} [add_comm_group M] [module R M] [module S M] [is_scalar_tower R S M]
+
+/-- For a `R`-derivation `S → M`, this is the map `S ⊗[R] S →ₗ[S] M` sending `s ⊗ₜ t ↦ s • D t`. -/
+def derivation.tensor_product_to (d : derivation R S M) : S ⊗[R] S →ₗ[S] M :=
```

Comment:
I like it that you used `D` in the docstring. Why not use it in the actual type as well? (Also elsewhere.)
```suggestion
/-- For a `R`-derivation `D : S → M`, this is the map `S ⊗[R] S →ₗ[S] M` sending `s ⊗ₜ t ↦ s • D t`. -/
def derivation.tensor_product_to (D : derivation R S M) : S ⊗[R] S →ₗ[S] M :=
```